src/wasm/wasm-type.cpp

```diff
@@ -155,18 +163,55 @@ struct TypePrinter {
   std::ostream& print(const Rtt& rtt);
 
 private:
-  template<typename T, typename F> std::ostream& printChild(T curr, F printer) {
-    auto it = depths.find(curr.getID());
-    if (it != depths.end()) {
-      assert(it->second <= currDepth);
-      size_t relativeDepth = currDepth - it->second;
-      return os << "..." << relativeDepth;
-    }
-    depths[curr.getID()] = ++currDepth;
-    printer();
-    depths.erase(curr.getID());
-    return os;
-  }
+  template<typename T, typename F> std::ostream& printChild(T curr, F printer);
+};
+
+// Helper for hashing the structures of TypeInfos and HeapTypeInfos. Uses de
+// Bruijn indices to represent back edges so that infos referring to different
+// type IDs but sharing a finite structure will compare and hash the same.
+struct TypeStructureHasher {
+  bool topLevelOnly;
+  size_t currDepth = 0;
```

Comment:
It looks like currDepth is in the de Bruijn sense, and not in the tree sense, is that right? As there is no decrement for it. If so a comment could help as the same name is used in the printer where it is in the tree sense. Or maybe a different name? https://en.wikipedia.org/wiki/De_Bruijn_index doesn't use "depth", so perhaps `currIndex`?

The `erase` where a decrement would appear (line 1378) also implies this index in a particular subtree, is that right?

Reply:
Yeah, that was a bug I think. But this should be fixed by the latest change.